File: Python/0896.py
class Solution:
    def isMonotonic(self, nums: list[int]) -> bool:
        if len(nums) == 1:
            return True

        increase = False
        decrease = False
        current = nums[0]

        for i in range(1, len(nums)):
            if nums[i] < current:
                decrease = True
            if nums[i] > current:
                increase = True
            if increase and decrease:
                return False

            current = nums[i]

        return True

# Time complexity: O(n)
# Space complexity: O(1)

File: C++/0896.cpp
#include <vector>

class Solution {
public:
    bool isMonotonic(std::vector<int>& nums) {
        if (nums.size() == 1) return true;
        
        bool increase = 0, decrease = 0;
        int current = nums[0];

        for(int i = 1; i < nums.size(); i++){
            if (nums[i] < current) increase = 1;
            if (nums[i] > current) decrease = 1;
            if (increase && decrease) return false;
            current = nums[i];
        }
        return true;
    }
};
// Time Complexity: O(n)
// Space Complexity: O(1)